doc/adr/0009-time-graph-search-functionality.md

```diff
@@ -0,0 +1,154 @@
+```
+{
+  "parameters": {
+    "requested_timerange": {
+      "start": 111111111,
+      "end": 222222222,
+      "nbTimes": 1920
+    },
+    "requested_items": [
+      1,
+      2
+    ],
+    "regex_map_filters": {1: ["openat"]},
+    "full_search": true
+  }
+}
+```
+-   `regex_map_filters` is an array with requests for searching. The  key  of  this  map  can  be \"1\" (means  DIMMED) or \"4\" (means  EXCLUDED) and  the  value  is  an  array  of  the  desired  search  query (e.g. {\"1\": [\"openat\", \"duration>10ms\"]})". The time event filtering uses its own **domain-specific language (DSL)**, which supports various operators and filter expressions. The definition of regexes is outside the scope of this ADR.
+-  The `full_search` key is an optional parameter that enables the full_seach (deep search) or not. 
+   - If true then the data provider is expected to perform a full_search of elements that occur between the usual elements found at the requested times (e.g. in gaps).  If any element matches the specified regex filter, at least one such  matching element per gap should be returned. It is not expected to return  all matching elements found in a single gap, for performance reasons.  
+   - On the client side when we try to perform a graph search we could first do a normal non-full_search query for all the visible rows, in order to quickly show something to the user. This would be followed by a full_search in the background. Because this process can be very slow, we could do it one row at a time and update the graph incrementally as each row's fine-grained result comes back. There might be better ways to combine non-full_search with full_search, therefore the sequence of operations to get the best user experience should be in another ADR. 
+
+Here's an example Sequence Diagram for the search request above:
+
+```mermaid
+sequenceDiagram
+    actor User
+    participant timegraph as timegraph-output-component
+    participant server as Trace Server
+    User->>timegraph: enter search filter regex ("openat")
+    timegraph->>server: POST /experiments/{expUUID}/outputs/timegraph/{outputId}/states
+    Note over timegraph,server: requested_timerange=[current_timerange]
+    Note over timegraph,server: requested_items=[selected_entry_Ids]
+    Note over timegraph,server: regex_map_filters={"1": ["openat"]}
+    Note over timegraph,server: full_search=true
+    server->>timegraph: 200: TimelineChart.TimeGraphModel
+    timegraph->>timegraph: TimeGraphChartProviders.stateStyleProvider updates state style
+    timegraph->>timegraph: Update timegraph
+```
+
+I've attached a link to a prototype in trace-extension that makes use of the aforementioned parameters to  search time events in time graph views:
+- [https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886](https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886)
```

Comment:
This format works, but we tend to prefer listing links at the end of the document, below the last line of text (see ADR 0002-GraphQL for an example).

So here would be:
`- [https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886][pull-886]`

And at the end of the ADR:
`[pull-886]: https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/886`